### PR TITLE
pipboard: make improvements for configuring pipeline and creating.

### PR DESCRIPTION
**Pull request by yorkie**
Now to configuration a pipeline:

Screen Shot 2020-07-08 at 11 21 10 PM

And create a pipeline is from templates as:

Screen Shot 2020-07-08 at 11 22 57 PM

Still working in progress:

- [x] Daemon configuration
  - [x] config local plugins config file.
  - [ ] ~~upgrade daemon and pipboard.~~
- [x] Plugin manager
  - [x] clean
  - [x] install
- [x] Job/Pipeline view
  - [x] real-time pulling the job states.

### CODE REVIEW

**Comment by yorkie** [comment]
And this PR would add new params for all builtin plugins.

**Comment by FeelyChau** [comment]
If sdk test failed, please focus on:
https://github.com/alibaba/pipcook/pull/326

**Comment by yorkie** [comment]
Added Job View as:

Screen Shot 2020-07-13 at 9 37 29 PM

**Comment by yorkie** [comment]
Added the plugin manager UI:

Screen Shot 2020-07-14 at 6 21 12 PM

**Comment by yorkie** [comment]
Added setting UI:

Screen Shot 2020-07-14 at 7 31 10 PM

File: packages/pipboard/src/layouts/BasicLayout/index.jsx [hunk 1]
```diff
@@ -1,35 +1,79 @@
...
+
   render() {
     return (
       <div className="dashboard">
+        <Dialog
+          className="new-pipeline-dialog"
+          title="Create a new Pipeline"
+          visible={this.state.newPipelineDialogVisible}
+          onClose={this.onClosePipelineDialog}
+          onCancel={this.onClosePipelineDialog.bind(this, 'cancel')}
```
Comment (rickycao-qy) [inline packages/pipboard/src/layouts/BasicLayout/index.jsx:60]: Is it necessary to use bind?

File: packages/pipboard/src/utils/request.js [hunk 2]
```diff
@@ -10,10 +13,58 @@ const createGeneralRequest = (agent) => async (...args) => {
...
+
+    // register extra handlers.
+    Object.keys(handlers)
+      // handle `handlers.error` manually.
+      .filter((name) => name !== 'error')
+      .forEach((name) => {
+        es.addEventListener(name, handlers[name]);
+      });
+  });
+};
```
Comment (rickycao-qy) [inline packages/pipboard/src/utils/request.js:70]: EOF

File: packages/pipboard/src/pages/Plugin/index.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    cursor: pointer;
+  }
+  .plugin-item-actions {
+    margin-top: 10px;
+  }
+}
+
+.plugin-dialog {
+  width: 60%;
+}
```
Comment (rickycao-qy) [inline packages/pipboard/src/pages/Plugin/index.scss:25]: EOF

File: packages/pipboard/src/pages/Job/Detail/index.scss [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        }
+        button {
+          margin: 0 5px;
+        }
+      }
+    }
+  }
+  
+  
+}
```
Comment (rickycao-qy) [inline packages/pipboard/src/pages/Job/Detail/index.scss:109]: Ditto

File: packages/pipboard/src/pages/Pipeline/Detail/index.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        }
+        button {
+          margin: 0 5px;
+        }
+      }
+    }
+  }
+  
+  
+}
```
Comment (rickycao-qy) [inline packages/pipboard/src/pages/Pipeline/Detail/index.scss:85]: Ditto

**Comment by yorkie** [comment]
We will move this PR to imgcook/pipboard.

**Pull request closed without merging**